### PR TITLE
Move common code into magellan-library

### PR DESCRIPTION
**Pull request by narenkmanoharan**
This PR moves common code like the screencontainer and transitions into magellan-common so that we can share this across legacy and library.

### CODE REVIEW

File: magellan-library/build.gradle [hunk 1]
```diff
@@ -42,8 +42,7 @@ android {
 }
 
 dependencies {
-
-  implementation project(':magellan-legacy')
```
Comment (narenkmanoharan) [inline magellan-library/build.gradle:46]: no more dependency on legacy ✨

Comment (ryanmoelter) [reply magellan-library/build.gradle:46]: Why not move these into Magellan library and have legacy depend on that? That way we can avoid creating another module, and we're going to need that dependency when we make the compatibility navigator in legacy

Comment (narenkmanoharan) [reply magellan-library/build.gradle:46]: I was thinking we would make the compatibility navigator in magellan-support. But i suppose this works too.

Comment (narenkmanoharan) [reply magellan-library/build.gradle:46]: Done

**Pull request merged**